src/en/reference-hook-tools.md

```diff
@@ -0,0 +1,564 @@
+Whatever information you need to propagate for the remote charm to work must be
+propagated via relation-set, with the single exception of the `private-address`
+key, which is always set before the unit joins.
+
+You may wish to overwrite the `private-address` setting, for example if you're
+writing a charm that serves as a proxy for some external service; but you 
+should in general avoid _removing_ that key, because most charms expect that
+value to exist unconditionally.
+
+All values set are stored locally until the hook completes; at that point, if
+the hook exit code is 0, all changed values will be communicated to the rest of
+the system, causing -changed hooks to run in all related units.
+
+There is no way to write settings for any unit other than the local unit; but
+any hook on the local unit can write settings for any relation the local unit
+is participating in.
+
+
+### status-get
+
+The `status-get` hook tool allows a charm to query what is recorded in Juju as 
+the current workload status. Without arguments, it just prints the workload
+status value e.g. 'maintenance'. With `--include-data` specified, it prints
+YAML which contains the status value plus any data associated with the status.
+
+Examples:
+
+```no-highlight
+status-get
+status-get --include-data
+```
+
+
+### status-set
+
+The status mechanism allows Juju and its 
+charms to accurately reflect their current status. This places the 
+responsibility on the charm to know its status, and set it accordingly using
+the `status-set` hook tool.
+
+This hook tool takes 2 arguments. The first is the status to report, which can 
+be one of the following:
+
+  - maintenance (the unit is not currently providing a service, but expects to 
+    be soon, E.g. when first installing)
+  - blocked (the unit cannot continue without user input)
+  - waiting (the unit itself is not in error and requires no intervention, 
+    but it is not currently in service as it depends on some external factor, 
+    e.g. a service to which it is related is not running)
+  - active (This unit believes it is correctly offering all the services it is
+    primarily installed to provide)
+
+For more extensive explanations of these statuses, and other possible status
+values which may be set by Juju itself,
+[please see the status reference page](./reference-status.html).
+
+The second argument is a user-facing message, which will be displayed to any
+users viewing the status, and will also be visible in the status history. This
+can contain any useful information. 
+
+This status message provides valuable feedback to the user about what is
+happening. Changes in the status message are not broadcast to peers and
+counterpart units - they are for the benefit of humans only, so tools
+representing Juju services (e.g. the Juju GUI) should check occasionally 
+and be told the current status message.
+
+Spamming the status with many changes per second would not be welcome
```

Comment:
Comment: I found this phrasing odd: "would not be welcome". Maybe: "is not advisable because..."